var/spack/repos/builtin/packages/graphviz/package.py

```diff
@@ -90,6 +95,9 @@ class Graphviz(AutotoolsPackage):
     for b in tested_bindings + untested_bindings:
         depends_on('swig', when=b)
 
+    depends_on('cairo~X', when='+pangocairo')
+    depends_on('pango~X', when='+pangocairo')
```

Comment:
This works (once the package patches around a bug in the ghostscript plugin):

    spack install graphviz+pangocairo+libgd^cairo+X^pango+X

I did a little bit of digging and didn't see why the problem only surfaces for `+X`....

It builds both ways, so I'm as content as I can afford to be.